beam/core/src/main/java/cz/o2/proxima/beam/direct/io/DirectBatchUnboundedSource.java

```diff
@@ -190,7 +191,12 @@ private DirectBatchUnboundedSource(
 
     if (partitions.isEmpty()) {
       // round robin
-      List<Partition> parts = reader().getPartitions(startStamp, endStamp);
+      List<Partition> parts =
```

Comment:
👍 this should allow watermark to progress faster as the "same timestamp" partitions should be evenly distributed